File: sol/2196.cpp
struct Solution {
    TreeNode* createBinaryTree(vector<vector<int>>& descriptions) {
        TreeNode *created[100001] = {nullptr};
        int parent[100001] = {0};
        int rootval = -1;
        for (auto &rec : descriptions){
            parent[rec[1]] = rec[0];
            if (created[rec[0]] == nullptr){
                created[rec[0]] = new TreeNode(rec[0]);
            }
            if (created[rec[1]] == nullptr){
                created[rec[1]] = new TreeNode(rec[1]);
            }
            rootval = rec[0];
            if (rec[2]){
                created[rec[0]]->left = created[rec[1]];
            }  else {
                created[rec[0]]->right = created[rec[1]];
            }
        }
        while (parent[rootval] != 0){
            rootval = parent[rootval];
        }
        return created[rootval];
    }
};

File: sol/2196/2196.py
class Solution:
    def createBinaryTree(self, descriptions: List[List[int]]) -> Optional[TreeNode]:
        created = dict()
        parent = dict()
        rootval = -1
        for p, c, l in descriptions:
            parent[c] = p
            if p not in created:
                created[p] = TreeNode(p)
            if c not in created:
                created[c] = TreeNode(c)
            rootval = p
            if l:
                created[p].left = created[c]
            else:
                created[p].right = created[c]
        while rootval in parent:
            rootval = parent[rootval]
        return created[rootval]lass Solution:
    def createBinaryTree(self, descriptions: List[List[int]]) -> Optional[TreeNode]:
        created = dict()
        isChild = set()
        for p, c, l in descriptions:
            isChild.add(c)
            if p not in created:
                created[p] = TreeNode(p)
            if c not in created:
                created[c] = TreeNode(c)
            if l:
                created[p].left = created[c]
            else:
                created[p].right = created[c]
        return created[next(k for k in created if k not in isChild)]
